src/eradiate/scenes/measure/_core.py

```diff
@@ -70,12 +70,53 @@
 
 
 def _measure_spectral_config_srf_converter(value: t.Any) -> Spectrum:
+    """
+    Converter for :class:`MeasureSpectralConfig` ``srf`` attribute.
+
+    Parameters
+    ----------
+    value : Any
+        Value to convert.
+
+    Returns
+    -------
+    Spectrum
+        Converted value.
+
+    Notes
+    -----
+    The behaviour of this converter depends on the value type:
+    * If ``value`` is not a string, it is passed to the
+      :class:`.spectrum_factory`'s converter.
+    * If ``value`` is a path, the corresponding file is loaded
+    * If ``value`` is a string, it is interpreted as a SRF identifier:
+      * If the identifier does not end with `-raw`, the converter looks for a
+        prepared version of the SRF and loads it if it exists, else it loads the
+        raw SRF.
+      * If the identifier ends with `-raw`, the converter loads the raw SRF.
+
+    """
     if isinstance(value, (str, os.PathLike)):
-        ds = converters.load_dataset(
-            f"spectra/srf/{value}.nc"
-            if not str(value).endswith(".nc")  # keyword for file in data store
-            else value  # path to file (local or in data store)
-        )
+
+        # path (local or in data store)
+        if str(value).endswith(".nc"):
+            ds = converters.load_dataset(value)
+
+        # identifier for a file in the data store
+        if isinstance(value, str):
+
+            try:
+                ds = converters.load_dataset(f"spectra/srf/{value}.nc")
+            except DataError as e:
+                if value.endswith("-raw"):
+                    raise e
+                else:
+                    warnings.warn(
+                        f"Could not serve SRF '{value}' from the data store."
+                        f"Trying to serve '{value}-raw' instead."
+                    )
+                    ds = converters.load_dataset(f"spectra/srf/{value}-raw.nc")
```

Comment:
Okay, this is still unsafe and hard to debug due to an unpredictable control flow. Let's try something a little safer:
```suggestion
    ds = _load_srf_ds(value)
```
where 
```python
def _load_srf_ds(value):
    # path (local or in data store)
    if str(value).endswith(".nc"):
        return converters.load_dataset(value)

    # identifier for a file in the data store
    if isinstance(value, str):
        try:
            return converters.load_dataset(f"spectra/srf/{value}.nc")
        except DataError as e:
            if value.endswith("-raw"):
                raise e
            else:
                warnings.warn(
                    f"Could not serve SRF '{value}' from the data store."
                    f"Trying to serve '{value}-raw' instead."
                )
                return converters.load_dataset(f"spectra/srf/{value}-raw.nc")

    # this is an abnormal state
    raise ValueError(f"unhandled value '{value}'")
```
The utility function will exit as soon as valid data is loaded, which is much cleaner than skipping blocks using conditionals. The final safeguard ensures that unhandled values are reported (failure would be silent in your original implementation).

Reply:
I assumed you meant that statement to be under the top `if`:
```python
if isinstance(value, (str, os.PathLike)):
    ds = _load_srf_ds(value)
    ...
else:
    converter = spectrum_factory.converter(quantity="dimensionless")
    return converter(value)
```